internal/mode/static/state/change_processor.go

```diff
@@ -248,3 +249,9 @@ func (c *ChangeProcessorImpl) Process() (bool, *graph.Graph) {
 
 	return true, c.latestGraph
 }
+
+func (c *ChangeProcessorImpl) GetLatestGraph() *graph.Graph {
```

Comment:
I haven't added the tests yet, but does anyone know if this function doesn't do what I'd like it to do just from taking a look at it? (Passing the processor to the Collector, and calling this function to get the latest graph)?

Also, since I'm just reading from the graph, I probably don't need to Lock and Unlock right?

Reply:
> Also, since I'm just reading from the graph, I probably don't need to Lock and Unlock right?

the lock is needed, because another goroutine can write to c.latestGraph in parallel here https://github.com/nginxinc/nginx-gateway-fabric/blob/9dd2968cb329ad3df94e6978f1e5f6a892375630/internal/mode/static/state/change_processor.go#L241  , which is a data race -- https://go.dev/doc/articles/race_detector